src/IssueViz/FixSuggestion/FixSuggestionHandler.cs

```diff
@@ -19,15 +19,167 @@
+            logger,
+            documentNavigator,
+            issueSpanCalculator,
+            openInIdeConfigScopeValidator,
+            ideWindowService, 
+            fixSuggestionNotification)
+    {
+    }
+
+    internal FixSuggestionHandler(
+        IThreadHandling threadHandling,
+        ILogger logger,
+        IDocumentNavigator documentNavigator,
+        IIssueSpanCalculator issueSpanCalculator,
+        IOpenInIdeConfigScopeValidator openInIdeConfigScopeValidator,
+        IIDEWindowService ideWindowService, 
+        IFixSuggestionNotification fixSuggestionNotification)
+    {
+        this.threadHandling = threadHandling;
+        this.logger = logger;
+        this.documentNavigator = documentNavigator;
+        this.issueSpanCalculator = issueSpanCalculator;
+        this.openInIdeConfigScopeValidator = openInIdeConfigScopeValidator;
+        this.ideWindowService = ideWindowService;
+        this.fixSuggestionNotification = fixSuggestionNotification;
+    }
+
+    public void ApplyFixSuggestion(ShowFixSuggestionParams parameters)
+    {
+        try
+        {
+            logger.WriteLine(FixSuggestionResources.ProcessingRequest, parameters.configurationScopeId, parameters.fixSuggestion.suggestionId);
+            ideWindowService.BringToFront();
+            fixSuggestionNotification.ClearAsync().Forget();
+
+            if (!ValidateConfiguration(parameters.configurationScopeId, out var configurationScopeRoot, out var failureReason))
+            {
+                logger.WriteLine(FixSuggestionResources.GetConfigScopeRootPathFailed, parameters.configurationScopeId, failureReason);
+                fixSuggestionNotification.InvalidRequestAsync(failureReason).Forget();
+                return;
+            }
+
+            threadHandling.RunOnUIThread(() => ApplyAndShowAppliedFixSuggestions(parameters, configurationScopeRoot));
+            logger.WriteLine(FixSuggestionResources.DoneProcessingRequest, parameters.configurationScopeId, parameters.fixSuggestion.suggestionId);
+        }
+        catch (Exception exception) when (!ErrorHandler.IsCriticalException(exception))
+        {
+            logger.WriteLine(FixSuggestionResources.ProcessingRequestFailed, parameters.configurationScopeId, parameters.fixSuggestion.suggestionId, exception.Message);
+        }
+    }
+
+    private bool ValidateConfiguration(string configurationScopeId, out string configurationScopeRoot, out string failureReason)
+    {
+        return openInIdeConfigScopeValidator.TryGetConfigurationScopeRoot(configurationScopeId, out configurationScopeRoot, out failureReason);
+    }
+
+    private void ApplyAndShowAppliedFixSuggestions(ShowFixSuggestionParams parameters, string configurationScopeRoot)
+    {
+        var absoluteFilePath = Path.Combine(configurationScopeRoot, parameters.fixSuggestion.fileEdit.idePath);
+        var textView = GetFileContent(parameters, absoluteFilePath);
+        if (!ValidateFileExists(textView, absoluteFilePath))
```

Comment:
Minor:
The `GetFileContent` and the `ValidateFileExists` should most of the times be called together. They could be combined in a single method, as we also do for the `ValidateConfiguration`.
Something like: `ValidateFileExists(absoluteFilePath, out var textView)`

Reply:
I disagree. Getting the file content and validating if it exists should be separate, otherwise it violates the SRP (single responsibility principle).

The same with ValidateFileExists(absoluteFilePath, out var textView). The out parameter is already an indicator that the method does more than it should: validating and returning something. 

I will leave them separate for now.